app/bundles/LeadBundle/Entity/LeadListRepository.php

```diff
@@ -914,6 +914,38 @@ public function getListFilterExpr($filters, &$parameters, QueryBuilder $q, $not
                         $q->expr()->comparison('l.id', $func, sprintf('(%s)', $subqb->getSQL()))
                     );
                     break;
+                case 'lead_email_received':
+                    $ignoreAutoFilter = true;
+
+                    $func = (in_array($func, array('neq', 'notIn'))) ? 'NOT IN' : 'IN';
+                    // Special handling of lead lists
+                    foreach ($details['filter'] as &$value) {
+                        $value = (int) $value;
+                    }
+
+                    // Generate a unique alias
+                    $alias = $this->generateRandomParameterName();
+
+                    $subqb = $this->_em->getConnection()->createQueryBuilder();
+                    $subqb->select($alias . '.lead_id')
+                        ->from(MAUTIC_TABLE_PREFIX.'email_stats', $alias)
+                        ->where(
+                            $subqb->expr()->in($alias . '.email_id', $details['filter'])
+                        )
+                        ->andWhere(
+                            $subqb->expr()->eq($alias.'.is_read', true));
```

Comment:
PostgreSQL will actually choke on the boolean due to how Doctrine translates the query. When assigning boolean values, you have to assign it as a parameter like:

```
$q->where(
    $q->expr()->ex('alias.field', ':parameterPlaceholder')
)
 ->setParameter('parameterPlaceholder', true, 'boolean');
```

In this case where the query is dynamic and no single parameter is guaranteed, you have to create a unique parameter for it. I think something like this will work (untested):
```
$parameter2 = $this->generateRandomParameterName();
$subqb->andWhere(
    $subqb->expr()->eq($alias.'.is_read', ":$parameter2") // note the colon before $parameter2
)
    ->setParameter($parameter2, true, 'boolean'); // no colon here
```